protocols/interfaces.go

```diff
@@ -50,6 +51,11 @@ type ObjectiveEvent struct {
 	BlockNum           uint64
 }
 
+// Storable is an object that can be stored by the store.
+type Storable interface {
+	json.Marshaler
+}
```

Comment:
Possibly?

```suggestion
type Storable interface {
	json.Marshaler
	// Related returns a slice of related objects that need to be stored along with the objective
	Related() []Storable
}
```

End-of-the-line `Storable` entities (channels, consensus_channels) would have to implement a noop something like: 

```
func (c Channel) Related() []protocols.Storable {
 return []protocols.Storable{} 
}
```

Reply:
The current situation is that objectives have related objects (Channels, and now ConsensusChannels). These related objects have no related objects -- if they did, your suggestion might make sense.

My opinion is,  as a first line tactic, we should reduce the number of relationships that must be stored together (because that is complex). So, we should avoid adding related entities to `Channels` & `ConsensusChannels` unless it's actually necessary.

My fear is that we would slowly build a bespoke ORM with this kind of work.